lib/bloc_record/persistence.rb

```diff
@@ -22,6 +22,76 @@ def create(attrs)
       new(data)
     end
 
+    def update(ids, updates)
+      # add a check for class of updates in case an array of attributes for diff
+      # records is passed (assignment 5)
+      if updates.class == Hash
+        updates = BlocRecord::Utility.convert_keys(updates)
+        updates.delete("id")
+        updates_array = updates.map { |key, value| "#{key}=#{BlocRecord::Utility.sql_strings(value)}" }
+      elsif updates.class == Array
+        updates_array = updates
+      end
+
+      if ids.class == Fixnum
+        where_clause = "WHERE id = #{ids};"
+      elsif ids.class == Array
+        where_clause = ids.empty? ? ";" : "WHERE id IN (#{ids.join(",")});"
+      else
+        where_clause = ";"
+      end
+
+      connection.execute <<-SQL
+        UPDATE #{table}
+        SET #{updates_array * ","} #{where_clause}
+      SQL
+
+      true
+    end
+
+    def destroy(*id)
+      if id.length > 1
+        where_clause = "WHERE id IN (#{id.join(",")})"
+      else
+        where_clause = "WHERE id = #{id.first};"
+      end
+
+      connection.execute <<-SQL
+        DELETE FROM #{table}
+        #{where_clause}
+      SQL
+
+      true
+    end
+
+    def destroy_all(conditions=nil)
+      if conditions.class == Hash && !conditions.empty?
+        conditions = BlocRecord::Utility.convert_keys(conditions)
+        conditions = conditions.map {|key, value| "#{key} = #{BlocRecord::Utility.sql_strings(value)}"}.join(" and ")
+
+        connection.execute <<-SQL
+          DELETE FROM #{table}
+          WHERE #{conditions}
+        SQL
+      elsif conditions.class == String && !conditions.empty?
+        if conditions.length > 1
```

Comment:
There are two cases where the first argument is a String. One of them is bullet point three.